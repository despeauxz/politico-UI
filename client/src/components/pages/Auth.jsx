import React, { Component, Fragment } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import SignupForm from '@components/shared/Form/SignupForm';
import LoginFrom from '@components/shared/Form/LoginForm';
import Header from '@components/shared/Header';
import { renderAuthPropTypes } from '@helpers/proptypes';

class Auth extends Component {
    static propTypes = {
        type: PropTypes.string.isRequired
    };

    static defaultProps = {
        type: ''
    };

    getFormTitle = () => {
        switch (this.props.type) {
        case 'signup':
            return 'Get Registered Today';
        default:
            return 'Welcome Back';
        }
    }

    getFormField = () => {
        switch (this.props.type) {
        case 'signup':
            return <SignupForm />;
        default:
            return <LoginFrom />;
        }
    }

    render() {
        const title = this.getFormTitle();
        const field = this.getFormField();
        const { from } = this.props.location.state ? this.props.location.state : { from: { pathname: '/' } };
        const newLocation = this.props.type === 'login' ? from : '/';

        if (this.props.isAuthenticated) {
            return <Redirect to={newLocation} />;
        }
        
        return (
            <Fragment>
                <Header />
                <div className="container mx-auto">

                    <div className="w-3/4 mx-auto mt-8 sm:w-11/12 md:w-3/5 lg:w-3/5 xl:w-1/2">
                        <div className="text-center">
                            <h2 className="text-md font-bold mb-10">{ title }</h2>
                            <hr />
                            
                            { field }
                        </div>
                    </div>
                </div>

            </Fragment>
        );
    }
}

Auth.propTypes = {
    ...renderAuthPropTypes,
};

Auth.defaultProps = {
    type: null
};

export default Auth;
