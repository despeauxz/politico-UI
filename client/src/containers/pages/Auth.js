import { connect } from 'react-redux';
import AuthComponent from '../../components/pages/Auth';

const mapStateToProps = state => ({
    isLoading: state.auth.loading,
    isAuthenticated: state.auth.isAuthenticated,
    errors: state.auth.errors
});

export default connect(mapStateToProps)(AuthComponent);