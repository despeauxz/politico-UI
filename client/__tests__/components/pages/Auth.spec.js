import React from 'react';
import Auth from '@components/pages/Auth';


describe('Auth Pages', () => {
    it('should render the sign up page based on props ~ type', () => {
        const shallowWrapper = shallow(<Auth type="signup" />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('SignupForm')).toBeTruthy();
    });

    it('should render the login page based on props ~ type', () => {
        const shallowWrapper = shallow(<Auth type="login" />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('LoginForm')).toBeTruthy();
    });
});
