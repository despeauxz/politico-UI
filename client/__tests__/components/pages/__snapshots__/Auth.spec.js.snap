// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Pages should render the login page based on props ~ type 1`] = `
<Fragment>
  <div
    className="container"
  >
    <HomeHeader />
    <div
      className="main-wrapper auth-wrapper"
    >
      <div
        className="auth text-center"
      >
        <h2>
          Welcome Back
        </h2>
        <hr />
        <Connect(withRouter(LoginFrom)) />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Auth Pages should render the sign up page based on props ~ type 1`] = `
<Fragment>
  <div
    className="container"
  >
    <HomeHeader />
    <div
      className="main-wrapper auth-wrapper"
    >
      <div
        className="auth text-center"
      >
        <h2>
          Get Registered Today
        </h2>
        <hr />
        <Connect(withRouter(SignupForm)) />
      </div>
    </div>
  </div>
</Fragment>
`;
